.tests/tests/jupytertest.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { test, expect,  } from '@playwright/test';
+import { EnvPool } from '../environments';
+import * as fs from 'fs';
+
+const CSAE_ENV_PASSWORD = process.env.CSAE_ENV_PASSWORD || 'asdfasdf';
+const CSAE_WORKERS_COUNT = parseInt(process.env.CSAE_WORKERS_COUNT || '1');
+const CSAE_PARALLEL_TESTS_COUNT = parseInt(process.env.CSAE_PARALLEL_TESTS_COUNT || '1');
+const envPool = new EnvPool(Math.floor(CSAE_WORKERS_COUNT/CSAE_PARALLEL_TESTS_COUNT));
+
+const  CSAE_CI_JOB_IDX = parseInt(process.env.CSAE_CI_JOB_IDX || '0');
+const CSAE_CI_JOB_COUNT= parseInt(process.env.CSAE_CI_JOB_COUNT || '1');
+
+const sleep = (ms) => new Promise(r => setTimeout(r, ms));
+
+test.describe.configure({ mode: 'parallel' });
+// files.txt will contain the list of files to be tested.
+// It is generated by the following command:
+// find .. -name '*.ipynb' | grep 'Getting_Started/' > ./files.txt
+// find .. -name '*.ipynb'  > ./files.txt
+const FILE_NAME = './files.txt';
+const SKIPFILE_NAME = './skip_files.txt';
+function readFileIntoArray(filename) {
+    try {
+        const data = fs.readFileSync(filename, 'utf8');
+        const lines = data.split('\n').map((line) => line.trim()).filter((line) => line !== '');
+        return lines;
+    } catch (err) {
+        console.error(`Error reading file '${FILE_NAME}': ${err.message}`);
+        return [];
+    }
+}
+
+const skipfiles = readFileIntoArray(SKIPFILE_NAME)
+const files = readFileIntoArray(FILE_NAME).filter((name) => skipfiles.indexOf(name) === -1);
+
+const testCount = Math.ceil(files.length / CSAE_CI_JOB_COUNT);
+
+for (let i = 0; i < testCount; i++) {
+    const idx = i * CSAE_CI_JOB_COUNT + CSAE_CI_JOB_IDX;
+    if(idx >= files.length) {
+        break;
+    }
+
+    const name = files[idx];
+    if(name === '') {
+        continue;
+    }
+    test(`test ${i}: ${name}`, async ({page},testInfo) => {
+        test.setTimeout(10800000);
+
+        // Create Env or get existing Env
+        const env = await envPool.getEnv(testInfo.parallelIndex);
+        
+        await sleep(5000);
```

Comment:
We should avoid arbitrary sleeps. Is there not a `waitFor()`-based logic that we could use here?